src/chuchichaestli/metrics/backbones/clip.py

```diff
@@ -0,0 +1,87 @@
+import torch
+import open_clip
+from torchvision.transforms import Normalize, InterpolationMode
+import torchvision.transforms.functional as TF
+import torch.nn.functional as F
+from torch.utils.checkpoint import checkpoint
+
+from chuchichaestli.metrics.backbones.backbone_encoder import Encoder
+from chuchichaestli.metrics.backbones.resizer import pil_resize
+
+ARCH_WEIGHT_DEFAULTS = {
+    "ViT-B-32": "laion2b_s34b_b79k",
+    "ViT-B-16": "laion2b_s34b_b88k",
+    "ViT-L-14": "datacomp_xl_s13b_b90k",
+    "ViT-bigG-14": "laion2b_s39b_b160k",
+}
+
+
+class CLIPEncoder(Encoder):
+    def setup(
+        self,
+        arch: bool = None,
+        pretrained_weights: bool = None,
+        clean_resize: bool = False,
+        depth: int = 0,
+    ):
+
+        if arch is None:
+            arch = "ViT-L-14"
+        if pretrained_weights is None:
+            pretrained_weights = ARCH_WEIGHT_DEFAULTS[arch]
+
+        self.model = open_clip.create_model(arch, pretrained_weights)
+        self.clean_resize = clean_resize
+        self.depth = depth
+
+    def transform(self, image):
+        mean = (0.48145466, 0.4578275, 0.40821073)
+        std = (0.26862954, 0.26130258, 0.27577711)
```

Comment:
Where do these numbers come from? Can you comment above them and make them configurable? I.e., use these values as default but make it possible to set different ones.

Reply:
Those are the infamous ImageNet mean and variances...